data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/events/EventSubscriber.java

```diff
@@ -34,26 +36,32 @@
   private final Runnable closeCallback;
   private final int maxPendingEvents;
   private final AtomicBoolean processingQueue;
+  private final TimeProvider timeProvider;
+  private UInt64 lastEventSentSeconds;
   final AsyncRunner asyncRunner;
 
   public EventSubscriber(
       final List<String> eventTypes,
       final SseClient sseClient,
       final Runnable closeCallback,
       final AsyncRunner asyncRunner,
-      final int maxPendingEvents) {
+      final int maxPendingEvents,
+      final TimeProvider timeProvider) {
     this.eventTypes = EventType.getTopics(eventTypes);
     this.sseClient = sseClient;
     this.closeCallback = closeCallback;
     this.maxPendingEvents = maxPendingEvents;
+    this.timeProvider = timeProvider;
     this.queuedEvents = new ConcurrentLinkedQueue<>();
     this.processingQueue = new AtomicBoolean(false);
     this.asyncRunner = asyncRunner;
     this.sseClient.onClose(closeCallback);
+    lastEventSentSeconds = timeProvider.getTimeInSeconds();
   }
 
   public void onEvent(final EventType eventType, final EventSource message)
       throws JsonProcessingException {
+    keepAlive();
```

Comment:
I don't think this will be triggered at the right times.  We only generate a `head` event when there are blocks - if we have a number of empty slots in a row we might not have any events received. Pre-genesis we wouldn't have any events coming through at all.  So I suspect the keep alive check needs to be on an independent timer.

Reply:
this is on any event, not a head event. but i'll remove it and we can do a separate thread, but we might have to be a little careful so we don't wind up with n threaded timers etc... so it's probably complicated enough for its own ticket.